src/pages/VaultsPage/VaultRow.tsx

```diff
@@ -36,15 +36,32 @@ export function VaultRow({
 
   useBlockListener(updateBalance, [vault.id]);
 
+  useEffect(() => {
+    const updateOwner = async () => {
+      const currentVault = contracts?.getVault(vault.vault_address);
+      if (!currentVault || !contracts) {
+        setUserIsOwner(false);
+        return;
+      }
+      const [ownerAddress, userAddress] = await Promise.all([
+        currentVault.owner(),
+        contracts.getMyAddress(),
+      ]);
+      setUserIsOwner(ownerAddress.toLowerCase() === userAddress.toLowerCase());
+    };
+    updateOwner();
+  }, [contracts, vault.id]);
+
```

Comment:
@levity it's easier to look at the net diff for your remaining comment than the commit diff